workflow/rules/long_read.smk

```diff
@@ -119,40 +97,50 @@ rule canu_correct:
         "../envs/canu.yaml"
     threads: 16
     shell:
-        "( if [ -d {params.outdir} ]; then rm -Rf {params.outdir}; fi &&"
-        " canu -correct -nanopore {input} -p {wildcards.sample} -d {params.outdir} genomeSize=30k minOverlapLength=10 minReadLength=200"
-        " useGrid=false {params.for_testing}"
-        " corMMapMerSize=10 corOutCoverage=50000 corMinCoverage=0 maxInputCoverage=20000"
-        " corOverlapper=minimap utgOverlapper=minimap obtOverlapper=minimap"
-        " corConcurrency={params.concurrency}"
-        " cormhapConcurrency={params.concurrency} cormhapThreads={params.concurrency}"
-        " cormmapConcurrency={params.concurrency} cormmapThreads={params.concurrency}"
-        " obtmmapConcurrency={params.concurrency} obtmmapThreads={params.concurrency}"
-        " utgmmapConcurrency={params.concurrency} utgmmapThreads={params.concurrency}"
-        " redConcurrency={params.concurrency} redThreads={params.concurrency}"
-        " ovbConcurrency={params.concurrency}"
-        " ovsConcurrency={params.concurrency}"
-        " oeaConcurrency={params.concurrency})"
-        "> {log} 2>&1"
-
-
-# rule medaka_consensus_reference:
-use rule assembly_polishing_ont as medaka_consensus_reference with:
+        """
+        ( if [ -d {params.outdir} ]; then rm -Rf {params.outdir}; fi &&
+        canu -correct -nanopore {input} -p {wildcards.sample} -d {params.outdir} genomeSize=30k minOverlapLength=10 minReadLength=200 \
+        useGrid=false {params.for_testing} \
+        corMMapMerSize=10 corOutCoverage=50000 corMinCoverage=0 maxInputCoverage=20000 \
+        corOverlapper=minimap utgOverlapper=minimap obtOverlapper=minimap \
+        corConcurrency={params.concurrency} \
+        cormhapConcurrency={params.concurrency} cormhapThreads={params.concurrency} \
+        cormmapConcurrency={params.concurrency} cormmapThreads={params.concurrency} \
+        obtmmapConcurrency={params.concurrency} obtmmapThreads={params.concurrency} \
+        utgmmapConcurrency={params.concurrency} utgmmapThreads={params.concurrency} \
+        redConcurrency={params.concurrency} redThreads={params.concurrency} \
+        ovbConcurrency={params.concurrency} \
+        ovsConcurrency={params.concurrency} \
+        oeaConcurrency={params.concurrency})
+        gzip -d {output.corr_gz}
+        2> {log}
+        """
+
+
+rule clip_adbc_corrected:
```

Comment:
Just out of curiosity: What does `adbc` mean? :)

Reply:
:) Sorry, actually hated that name myself, but somehow couldn't squeeze out anything better at this moment and didn't want to get hung. its a bad abbreviation for "clip adapters and barcodes" because here notramp is configured to only cut away those two (--incl_prim flag), but not the primers as opposed to in the downsample_and_trim_raw rule, where all three are clipped (default). The rational to keep the primers here, is to give the de-novo assembly at least a small chance to find overlaps, because they are tiny already with primers. Without primers de-novo assembly will be very precarious because often overlaps will be <10 bp. Doomed to fail. Of course including primers may mask variants, which is certainly a major drawback. But as I have mentioned before and elswhere already too, I think de-novo assembly anyway isn´t a valid approach with this type of amplicon-data, since by using primers we already dictated the way the target segments/amplicons have to be arranged in the genome in order to get data at all. Thus we make an a priori assumption about how the genome is structured and if that is not true the approach fails anyway. Therefore I don´t see how we can really gain any valuable information from de-novo assembly in addition to the polishing/variant-calling approach other than that we can say we have a second method. I still included read-correction, because it is my impression that a second approach was wanted, but I think it is worth discussing to entirely drop the de-novo assembly.